### PR TITLE
[ASAN][AMDGPU] NFC. Add CodeGen tests.

### PR DESCRIPTION
**Pull request by vpykhtin**
To show upcoming patch changes.

### CODE REVIEW

File: llvm/test/Instrumentation/AddressSanitizer/AMDGPU/asan_codegen_loop.ll [hunk 1]
```diff
@@ -0,0 +1,504 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 3
+; RUN: opt  -passes=asan -S < %s | FileCheck %s --check-prefix=OPT
+; RUN: opt < %s -passes='asan,default<O3>' -o - | llc -O3 -mtriple=amdgcn-hsa-amdhsa -mcpu=gfx90a -o - | FileCheck %s --check-prefix=LLC-W64
```
Comment (vitalybuka) [inline llvm/test/Instrumentation/AddressSanitizer/AMDGPU/asan_codegen_loop.ll:3]: Is llc test is important?

I assume you can do IR only test, so it's in Instrumentation

llc should be on CodeGen side, but I am not convinced that they are needed.

Comment (vitalybuka) [reply llvm/test/Instrumentation/AddressSanitizer/AMDGPU/asan_codegen_loop.ll:3]: I am quite confident that llc tests are unnecessary.
You test should reflect what are you going to change in code.

And the change is about
```
const char kAMDGPUBallotName[] = "llvm.amdgcn.ballot.i64";
const char kAMDGPUUnreachableName[] = "llvm.amdgcn.unreachable";
```

which are visible in IR after https://github.com/llvm/llvm-project/pull/72247

**Review by vitalybuka — APPROVED** [review]
LGTM if LLC is removed

**Comment by vpykhtin** [comment]
Ok, but cases in my tests are already covered by existing tests, so I'll better drop this PR and remove it from https://github.com/llvm/llvm-project/pull/72247.

**Pull request closed without merging**